src/DotVVM.Framework/Controls/SelectorBase.cs

```diff
@@ -104,5 +109,29 @@ public IValueBinding ItemTitleBinding
         }
         public static readonly DotvvmProperty ItemTitleBindingProperty =
             DotvvmProperty.Register<IValueBinding, SelectorBase>(nameof(ItemTitleBinding));
+
+        [ControlUsageValidator]
+        public static IEnumerable<ControlUsageError> ValidateUsage(ResolvedControl control)
+        {
+            if (control.Properties.ContainsKey(SelectorBase.ItemValueBindingProperty) &&
+                control.Properties.GetValue(SelectorBase.ItemValueBindingProperty) is ResolvedPropertyBinding valueBinding)
+            {
+                var t = valueBinding.Binding.ResultType;
+                if (!t.IsPrimitiveTypeDescriptor())
```

Comment:
You also actually don't need the `IsPrimitiveTypeDescriptor` method implemented on `ITypeDescriptor` since `valueBinding.Binding.ResultType` will always be `ResolvedTypeDescriptor`.